roadmap/propagate-rust-panic-through-native-frame.md

```diff
@@ -46,3 +48,37 @@ cases (contains destructors, no affordances). We will also have to
 specify this on a per-target basis, as the details will vary depending
 on what exception mechanism is in use.
 
+Another consideration is whether the the native frames include a `catch` block.
+There are several possibilities here as well:
+
+* "catch native" -- e.g. a C++ block catching a specific subset of C++ exceptions
+* "catch all" -- e.g. a C++ catch without an exception argument
+* "catch and release" -- e.g. a C++ "catch all" that re-`throw`s the exception
+
+So far we do **not** have plans to specify these cases.
```

Comment:
The reason I stated it this way is that the behavior of foreign code interacting with a Rust `panic` is of course dependent on parts of the `panic` ABI that otherwise do not need to be specified.

I mentioned this in Zulip, but I believe one such ABI element is the personality function. I don't believe any other verbiage in this document so far implies any constraints on the personality function, so I think it's good to be explicit here.

Reply:
> The reason I stated it this way is that the behavior of foreign code interacting with a Rust panic is of course dependent on parts of the panic ABI that otherwise do not need to be specified.

The Rust panic ABI does not interact with whatever unwinding ABI "C unwind" uses. At the "C unwind" boundary we translate Rust panics into a different kind of panic, e.g.,  a "platform Rust panic", and foreign code only interacts with those, not with Rust panics themselves.

Reply:
The only thing that needs to be possible is this translation, but this is always possible AFAICT, because we can always use `catch_unwind` with Rust panics, and after catching the panic, we can do whatever we want with it to make it conform to the native ABI.